src/databricks/sql/utils.py

```diff
@@ -134,7 +134,7 @@ def __init__(
         schema_bytes,
         max_download_threads: int,
         start_row_offset: int = 0,
-        result_links: List[TSparkArrowResultLink] = None,
+        result_links: List[TSparkArrowResultLink] = [],
```

Comment:
I'm a bint uncertain about this change. In JS I would just approve it 🙂  But in Python, default values are created once and thus are shared between function calls. So if one day somebody will decide to modify `self.result_links` (or other place where it is passed to) without paying extra attention - it will cause udesired side effects. Maybe better keep a default value `None` and change the type to `Optional[List[TSparkArrowResultLink]]`. WDYT?

Reply:
Boy, I learn something surprising about Python every day...!

I agree that [] is a bad default value, then. `Optional[List[TSparkArrowResultLink]]` is a fine type for the parameter to have. The logic for dealing with it being None, which I removed, would have to be put back in. (The code right below that code would also, I think, have to be indented, to go within the `if not None` if statement—that code being outside that if statement was the error mypy was originally upset about which caused me to touch this code at all, I think.)

For your consideration: if you want to get rid of the `= None` altogether, you can. There would be no default for that argument, and it would become a required parameter. It would be moved to before start_row_offset (as it would now be a positional parameter). I grepped the example code and didn't see any example usages in which the parameter it was omitted, but I don't know enough about the code to say if it should be optional.

So, either of those options seem fine to me.

Reply:
If you don't mind - let's keep `None` as default value, with all those `if <var> is not None`

Reply:
I have now done so.